Tests/Delphi.Mocks.Tests.Utils.pas

```diff
@@ -113,6 +150,357 @@ procedure TTestTValue.Test_TValue_NotEquals_Strings;
   CheckFalse(v1.Equals(v2));
 end;
 
+
+procedure TTestTValue.Test_TValue_Equals_Variants;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := 'hello';
+  var2 := 'hello';
+  v1 := TValue.FromVariant( var1 );
+  v2 := TValue.FromVariant( var2 );
+  CheckTrue(v1.Equals(v2));
+end;
+
+procedure TTestTValue.Test_TValue_Equals_Variants_BothEmpty;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := varEmpty;
```

Comment:
From the name of the test I assume that you tried to create an empty Variant. But what you are doing here actually is assign varEmpty (which is 0) to a Variant. The Unassigned function is what returns a Variant that has VType = varEmpty. So this test is defect and redundant with Test_TValue_Equals_Variants_BothUnAssigned .